libtenzir/src/tql2/eval_impl.cpp

```diff
@@ -254,35 +253,44 @@ auto evaluator::eval(const ast::index_expr& x) -> multi_series {
       return eval(ast::field_access{
         x.expr,
         location::unknown,
-        x.suppress_warnings,
+        x.has_question_mark,
         ast::identifier{*string, constant->source},
       });
     }
   }
   return map_series(
     eval(x.expr), eval(x.index),
     [&](series value, const series& index) -> multi_series {
+      const auto add_suppress_hint = [&](auto dh) {
+        if (x.rbracket != location::unknown) {
```

Comment:
Can you give an example why this check is here?

Reply:
I added a comment for this in 56acb971106.